src/main/kotlin/com/fasterxml/jackson/module/kotlin/Extensions.kt

```diff
@@ -48,4 +48,11 @@ inline fun <reified T : Any> SimpleModule.addSerializer(kClass: KClass<T>, seria
 inline fun <reified T : Any> SimpleModule.addDeserializer(kClass: KClass<T>, deserializer: JsonDeserializer<T>) = this.apply {
     addDeserializer(kClass.java, deserializer)
     addDeserializer(kClass.javaObjectType, deserializer)
+}
+
+inline fun <reified T> T.throwIfNullableTypeIsNull(): T {
+    if (null !is T && this == null) {
+        throw Exception()
```

Comment:
This is bit tricky because failure happens outside of call that has `DeserializationContext` -- usually you'd call `context.reportInputMismatch(...)` (or similar). That throws `MismatchedInputException`, for what that is worth.
That is used by databind. Might be worth checking how Kotlin module indicates errors however.
@dinomite and @viartemev can give more guidance.